### PR TITLE
bump for stdlib 9

### PR DESCRIPTION
**Pull request by sircubbi**
replaces usage of old validation_()-functions with proper type-definition and therefore allows the usage of `puppet-stdlib` 9.0.0.

Bonus: also fixes the CI ;)

### CODE REVIEW

**Review by djjudas21 — APPROVED** [review]
Looks good to me

File: manifests/module/detail.pp
Comment (nward) [inline manifests/module/detail.pp:12]: This Optional here has no effect - Optional means "can be undef, or ..." - but the default is `[]` so it is never undef.

The rest of the code here needs a tidy up where the defaults are not undef :)

File: manifests/sql.pp
Comment (nward) [inline manifests/sql.pp:5]: This should probably be `Stdlib::IP::Address::Nosubnet`

File: metadata.json
Comment (nward) [inline metadata.json:13]: This needs to be `>=4.25.0` for `Stdlib::IP::Address` etc.

**Comment by djjudas21** [comment]
Oops, was I a bit keen on the merge button?

**Comment by sircubbi** [comment]
Those are all valid points. I more or less just reimplemented whatever the validate_*()-functions were doing, but I guess it should be more precise and some of those `Optional`s can be removed indeed, since puppet has no way to reset a value to `undef` if it already has a default value.

I can tidy up all of that, I guess the best way would be to open a new PR, since this one is already merged? (Or would you rather fancy some force push to have everything in only one commit?)

**Comment by nward** [comment]
@djjudas21 oh woops I didn't realise you had merged it, I got an approval email not a merge haha.. 😬 

@sircubbi probably a new PR if that's ok? If you don't have time I can do one next day or so. Good work though, thanks!

**Pull request merged**
